modules/passkey/README.md

```diff
@@ -2,9 +2,33 @@
+
+The below sections give a high-level overview of the contracts present in this package.
+
+### [SafeWebAuthnSignerProxy](./contracts/SafeWebAuthnSignerProxy.sol)
+
+A proxy contract is uniquely deployed for each Passkey signer. The signer information i.e, Public key co-ordinates, Verifier address, Singleton address are immutable. All calls to the signer are forwarded to the `SafeWebAuthnSignerSingleton` contract.
+
+Use of `SafeWebAuthnSignerProxy` provides gas savings compared to the whole contract deployment for each signer creation. Both `SafeWebAuthnSignerProxy` and `SafeWebAuthnSignerSingleton` use no storage slots to avoid storage access violations defined in ERC-4337. The details on gas savings can be found in [this PR](https://github.com/safe-global/safe-modules/pull/370). This non-standard proxy contract appends signer information i.e., public key co-ordinates, and verifier data to the calldata before forwarding call to the singleton contract.
+
+### [SafeWebAuthnSignerSingleton](./contracts/SafeWebAuthnSignerSingleton.sol)
+
+`SafeWebAuthnSignerSingleton` contract is a singleton contract that implements the ERC-1271 interface to support signature verification, enabling signature data to be forwarded from a Safe to the `WebAuthn` library. This contract expects public key co-ordinates, and verifier address to be appended by the caller inspired from [ERC-2771](https://eips.ethereum.org/EIPS/eip-2771).
+
+### [SafeWebAuthnSignerFactory](./contracts/SafeWebAuthnSignerFactory.sol)
+
+The `SafeWebAuthnSignerFactory` contract deploys the `SafeWebAuthnSignerProxy` contract with the public key coordinates and verifier information. The factory contract also supports signature verification for the public key and signature information without deploying the signer contract, which is used during the validation of `UserOp`. Using [ISafeSignerFactory](./contracts/interfaces/ISafeSignerFactory.sol) interface and this factory contract address, new signers can be deployed.
+
+### [WebAuthn](./contracts/libraries/WebAuthn.sol)
+
+This library is used for generating signing message, hashing it and forwarding the call to the verifier contract. The `WebAuthn` library defines `Signature` struct containing `authenticatorData`, `clientDataFields`, followed by ECDSA signature's `r` and `s` components. The `authenticatorData` and `clientDataFields` required for generating the signing message. The `bytes` signature received in `verifySignature(...)` function is cast to `Signature` struct so, the caller has to take into account formatting the signature bytes as expected by the `WebAuthn` library.
+
```

Comment:
Can we document the expected format here? This is just an overview of the format, but I think we should be more precise and even specify a code snippet for example.

Reply:
https://github.com/safe-global/safe-modules/pull/394/commits/b0f5e39dd15a91973e0e0ade19fb6648e91b2db7